### PR TITLE
refactored metric collector tool.

### PR DESCRIPTION
**Pull request by ewiseblatt**
@lwander 

Segregated stackdriver stuff into its own modules to make it easier to add another option.
Support text or html so command line tool has readable output.
Other cleanups while I'm here.

### CODE REVIEW

File: google/stackdriver_monitoring/metric_collector.py [hunk 1]
```diff
@@ -69,27 +85,17 @@ def get_options():
   return parser.parse_args()
 
 
-def process_command(command, spectator, stackdriver, options):
+def process_command(command, registry):
```
Comment (lwander) [inline google/stackdriver_monitoring/metric_collector.py]: What is `registry` here?

File: google/stackdriver_monitoring/metric_collector_handlers.py [hunk 2]
```diff
@@ -1,24 +1,47 @@
...
+from datetime import datetime
+
+
+def accepts_html(request):
+  if not hasattr(request, 'headers'):
+    return False
+  accept = request.headers.get('accept', None)
+  if not accept:
+    return None
+  return 'text/html' in accept.split(',')
```
Comment (lwander) [inline google/stackdriver_monitoring/metric_collector_handlers.py]: This probably isn't sufficient if you look at all the ways your header can be formatted: https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html

File: google/stackdriver_monitoring/stackdriver_handlers.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        self.num_ok = 0
+
+      def handle_batch_response(self, index_str, good, bad):
+        index = int(index_str)
+        if bad:
+          self.batch_response[index] = 'ERROR {0}'.format(cgi.escape(str(bad)))
+          logging.error(bad)
+        else:
+          self.num_ok += 1
+          if not good:
```
Comment (lwander) [inline google/stackdriver_monitoring/stackdriver_handlers.py]: What's the difference between "bad", "ok", and "good" here?

**Comment by ewiseblatt** [comment]
updated

**Pull request merged**